src/main/kotlin/uk/gov/justice/digital/hmpps/incidentreporting/jpa/Report.kt

```diff
@@ -423,16 +424,20 @@ class Report(
       question = nomisQuestion.question,
     )
 
-  fun popLastQuestion(): Question? {
-    val lastQuestion = if (questions.isNotEmpty()) {
-      questions.last()
-    } else {
-      null
+  fun removeQuestion(question: Question) {
+    questions.remove(question)
+  }
+
+  fun removeQuestions(questionCodes: Set<String>) {
+    val questionsToRemove = questions.filter { questionCodes.contains(it.code) }
+    val questionCodesFound = questionsToRemove.mapTo(mutableSetOf()) { it.code }
+    val missingCodes = questionCodes - questionCodesFound
+    if (missingCodes.isNotEmpty()) {
+      throw QuestionsNotFoundException(missingCodes)
```

Comment:
What does this mean? Does it mean that if the client for whatever reason pass a question ID which is not answered it would get an error?

Is this necessary? Couldn't that just be ignored? I don't think there is any harm. I don't think I would pass anything extra but it shouldn't make any difference (e.g. it's a no-op to remove a question which was not answered before)